sw/host/opentitantool/src/command/otp.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::any::Any;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+
+use serde_annotate::{serialize, Annotate, Base};
+
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::otp::alert_handler::AlertRegs;
+use opentitanlib::otp::lc_state::LcStateVal;
+use opentitanlib::otp::otp_img::{OtpImg, OtpImgItem, OtpImgPartition, OtpImgValue};
+
+/// Generate CRC magic value for alert_handler configuration.
+#[derive(Debug, StructOpt)]
+pub struct AlertDigest {
+    #[structopt(help = "OTP memory map file containing alert_handler config in HJSON format")]
+    alert_cfg: PathBuf,
+    #[structopt(help = r#"Life cycle state ("TEST", "DEV", "PROD", PROD_END, "RMA")"#)]
+    lc_state: String,
```

Comment:
I'm not sure why we need this because OTP has entries for all applicable life cycle states. Can we calculate and print [them](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/otp_ctrl/data/otp_ctrl_mmap.hjson;drc=b003b78a2cf2f74bdbeb1f660163ee61623cad50;l=244) all?

Reply:
Could be done, but due to the way the bazel rule is setup it's easier to call into the tool for each lifecycle state.

For instance, a new overlay can be created with
```
otp_alert_digest(
    name = "otp_json_alert_digest_rma_cfg",
    otp_img = ":otp_json_owner_sw_cfg",
    lc_state = "RMA",
)
```

and then included in the OTP image for that lifecycle state using

```
otp_image(
    name = "img_rma",
    src = ":otp_json_rma",
    overlays = [
        ":otp_json_creator_sw_cfg",
        ":otp_json_owner_sw_cfg",
        ":otp_json_hw_cfg",
        ":otp_json_alert_digest_rma_cfg",
    ],
)
```

Reply:
Yes, for a single overlay, but the values of the remaining words would be wrong. Eventually, we will want to produce a final OTP image so I'd prefer updating them all at once.

Reply:
I see what you mean, I was thinking about this slightly wrong. Can do!

Reply:
Thanks!